hplsql/src/main/java/org/apache/hive/hplsql/Stmt.java

```diff
@@ -787,8 +787,19 @@ public Integer insertValues(HplsqlParser.Insert_stmtContext ctx) {
     for (int i = 0; i < rows; i++) {
       HplsqlParser.Insert_stmt_rowContext row =ctx.insert_stmt_rows().insert_stmt_row(i);
       int cols = row.expr().size();
-      for (int j = 0; j < cols; j++) {         
-        String value = evalPop(row.expr(j)).toSqlString();
+      for (int j = 0; j < cols; j++) {
+        Var var = evalPop(row.expr(j));
+        String value =  null;
+        if (var.type == Type.TIMESTAMP) {
+          value = String.format("TIMESTAMP '%s'", var.toString());
+        } else if (var.type == Type.DATE) {
+          value = String.format("DATE '%s'", var.toString());
+        } else {
+          value = var.toSqlString();
+        }
```

Comment:
How about `var.toString()`?

Reply:
As explained at https://github.com/apache/hive/pull/5182#discussion_r1570337451 comment, this format is required for insert also otherwise it throws exception. Example: insert into test values('Bob', 2024-04-17 10:20:30) throws exception here even the the type of second column is timestamp.

Reply:
Added Var.toSqlString(boolean isBuildSql) API and using the same.